sqlair.go

```diff
@@ -0,0 +1,172 @@
+package sqlair
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/canonical/sqlair/internal/expr"
+)
+
+// Statement is a sql statement with valid sqlair DML.
+type Statement struct {
+	pe *expr.PreparedExpr
+}
+
+// Query stores a database query and iterates over the results.
+type Query struct {
```

Comment:
Really minor thing, I think query _holds_ a database query? 'Stores' when dealing with database things might have the wrong implication. Also, as this is a type, I don't think that it 'iterates,' perhaps it 'allows iteration over results'?

Reply:
Changed.